app/workers/email_generation_worker.rb

```diff
@@ -1,21 +1,43 @@
+require "sidekiq-scheduler"
+
 class EmailGenerationWorker
   include Sidekiq::Worker
 
-  def perform(subscription_content_id)
-    subscription_content = SubscriptionContent.find(subscription_content_id)
+  sidekiq_options unique: :until_and_while_executing
 
-    email = Email.create_from_params!(email_params(subscription_content))
+  def perform
+    while subscription_contents.exists?
```

Comment:
If the plan is still to change over to running this on a timer rather than it being triggered by the SubscriptionContentWorker, I wonder whether we need the `while` here - we could just process everything where `email: nil` and not worry too much?